stingray/lightcurve.py

```diff
@@ -151,74 +183,146 @@ def __init__(self, time, counts, err=None, input_counts=True,
                 raise StingrayError("Statistic not recognized."
                                     "Please select one of these: ",
                                     "{}".format(valid_statistics))
-            if err_dist.lower() == 'poisson':
-                # Instead of the simple square root, we use confidence
-                # intervals (should be valid for low fluxes too)
+
+            if precise_poisson_err:
                 err = poisson_symmetrical_errors(counts)
-            else:
+
+            if not err_dist.lower() == 'poisson':
                 simon("Stingray only uses poisson err_dist at the moment, "
                       "We are setting your errors to zero. "
                       "Sorry for the inconvenience.")
                 err = np.zeros_like(counts)
 
         self.mjdref = mjdref
-        self.time = np.asarray(time)
-        dt_array = np.diff(np.sort(self.time))
-        dt_array_unsorted = np.diff(self.time)
-        unsorted = np.any(dt_array_unsorted < 0)
+        self.time = time
 
         if dt is None:
-            if unsorted:
-                logging.warning("The light curve is unordered! This may cause "
-                                "unexpected behaviour in some methods! Use "
-                                "sort() to order the light curve in time and "
-                                "check that the time resolution `dt` is "
-                                "calculated correctly!")
-
-            self.dt = np.median(dt_array)
-        else:
-            self.dt = dt
-
-        self.bin_lo = self.time - 0.5 * self.dt
-        self.bin_hi = self.time + 0.5 * self.dt
+            logging.warning("Computing the bin time ``dt``. This can take "
+                            "time. If you know the bin time, please specify it"
+                            " at light curve creation")
+            dt = np.median(np.diff(self.time))
 
+        self.dt = dt
         self.err_dist = err_dist
 
-        if unsorted:
-            self.tstart = np.min(self.time) - 0.5 * self.dt
-            self.tseg = np.max(self.time) - np.min(self.time) + self.dt
-        else:
-            self.tstart = self.time[0] - 0.5 * self.dt
-            self.tseg = self.time[-1] - self.time[0] + self.dt
-
-        self.gti = \
-            np.asarray(assign_value_if_none(gti,
-                                            [[self.tstart,
-                                              self.tstart + self.tseg]]))
-
-        check_gtis(self.gti)
-
-        good = create_gti_mask(self.time, self.gti, dt=self.dt)
+        self.tstart = self.time[0] - 0.5 * self.dt
+        self.tseg = self.time[-1] - self.time[0] + self.dt
```

Comment:
Have we checked that this is consistent with whatever `Lightcurve.make_lightcurve()` produces when the keywords `tstart` and `tseg` are set? This might be useful to do. :)

Reply:
I think that the line below from `make_lightcurve` gives the same result:
```
time = tstart + np.arange(0.5, 0.5 + len(counts)) * dt
```